src/test/ensemble_tests.cpp

```diff
@@ -1834,3 +1834,124 @@ TEST_F(EnsembleFlowTest, ErrorHandlingSkipsDeferredNodesExecutionIfExecutionFail
 
     EXPECT_EQ(pipeline.execute(), StatusCode::INVALID_SHAPE);
 }
+
+TEST_F(EnsembleFlowTest, ReloadPipelineDefinitionWithNewModelNameShouldPass) {
+    ConstructorEnabledModelManager managerWithDummyModel;
+    managerWithDummyModel.reloadModelWithVersions(config);
+
+    const std::string pipelineName = "originalName";
+    std::vector<NodeInfo> info{
+        {NodeKind::ENTRY, "request"},
+        {NodeKind::DL, "dummy_node", "dummy"},
+        {NodeKind::EXIT, "response"},
+    };
+    std::unordered_map<std::string, std::unordered_map<std::string, InputPairs>> connections;
+    // request O--------->O dummy node
+    connections["dummy_node"] = {
+        {"request", {{"pipeline_input_name", "b"}}}};
+    // dummy node O--------->O dummy node 1
+    connections["response"] = {
+        {"dummy_node", {{"a", "pipeline_output_name"}}}};
+    PipelineDefinition pd(pipelineName, info, connections);
+    auto status = pd.validate(managerWithDummyModel);
+    ASSERT_TRUE(status.ok());
+
+    config.setName("newDummy");
+    status = managerWithDummyModel.reloadModelWithVersions(config);
+    ASSERT_TRUE(status.ok()) << status.string();
+    std::vector<NodeInfo> infoNew{
+        {NodeKind::ENTRY, "request"},
+        {NodeKind::DL, "dummy_node", "newDummy"},
+        {NodeKind::EXIT, "response"},
+    };
+    status = pd.reload(managerWithDummyModel, infoNew, connections);
```

Comment:
How about testing `pd.validate()` aswell even though it is called inside `reload()`?

Reply:
Yes you are right - in case we just add model version with mapping we will only call validate not reload.

Reply:
Actually after a bit of thought I think that here calling reload is OK.
However there is a mistake in ReloadPipelineDefinitionWhen1ModelVersionBecomesAvailableShouldPass
since calling pipeline defintion reload wouldn't happen after all implementation - just validate.

Model versions would disappear, reappear and in this case there will be no reload - just revalidation.